Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionButton.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// Button for action displayed in `_ChatMessageActionsView`
+public typealias ChatMessageActionButton = _ChatMessageActionButton<NoExtraData>
+
+/// Button for action displayed in `_ChatMessageActionsView`
+open class _ChatMessageActionButton<ExtraData: ExtraDataTypes>: _Button, UIConfigProvider {
+    /// The data this view component shows.
+    public var content: ChatMessageActionItem? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    override public func defaultAppearance() {
+        super.defaultAppearance()
+        backgroundColor = uiConfig.colorPalette.background
+        titleLabel?.font = uiConfig.font.body
+        titleLabel?.adjustsFontForContentSizeCategory = true
+        contentEdgeInsets = UIEdgeInsets(top: 8, left: 16, bottom: 8, right: 32)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: -16)
+        contentHorizontalAlignment = .left
+    }
```

Comment:
Can we make this a bit nicer? Maybe rather than this we could just put the views into `ContainerView` and add it as a subview? We'd get all the margins and spacing for free there and it would match the rest of the app.

Also, it would be nice to improve the highlighted behavior, for example by changing the background of the button. We have a method in `UIConfig` for getting the proper color for the highlighted state.

Reply:
converted it to `UIControl`